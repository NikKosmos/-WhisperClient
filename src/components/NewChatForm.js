import React from 'react'
import PropTypes from 'prop-types'

class NewChatForm extends React.Component {

    state = {
        name: '',
        users: [],
    }

    onSubmit = (event) => {
        event.preventDefault()

        this.props.onSubmitNewChat(this.state.name, [this.props.currentUserId])

        this.setState({
            name: '',
            users: [],
        })
    }

    updateChatNameValue = (eventArg) => {
        this.setState({
            name: eventArg.target.value
        })
    }

    render() {
        return (
            <form onSubmit = { this.onSubmit }>
                <h1>Create new chat</h1>
                <input
                    name = 'chatName'
                    placeholder = 'chat name'
                    type = 'text'
                    value = { this.state.name }
                    onChange = { this.updateChatNameValue }
                />
                <button type = "submit">
                    Create
                </button>
            </form>
        )
    }
}

NewChatForm.propTypes = {
    currentUserId: PropTypes.string.isRequired,
    onSubmitNewChat: PropTypes.func.isRequired,
}

export { NewChatForm }