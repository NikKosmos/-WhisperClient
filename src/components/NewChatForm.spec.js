import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json';

import { NewChatForm } from './NewChatForm'

describe('NewChatForm', () => {
    
    const mockCurrentUserId = '0'     
    const mockOnSubmit = jest.fn()

    it ('should render correctly', () => {
        const testedComponent = shallow(<NewChatForm 
            onSubmitNewChat = { mockOnSubmit } currentUserId = { mockCurrentUserId }
        />)

        expect(shallowToJson(testedComponent)).toMatchSnapshot();
      })

})

describe('when typing chat name', () => {
    
    const mockCurrentUserId = '0'     
    const mockOnSubmit = jest.fn()
    const newTestChatName = 'Capitan Nemo Chat'

    const myComponent = shallow(<NewChatForm
        onSubmitNewChat = { mockOnSubmit } currentUserId = { mockCurrentUserId }
    />)

    beforeEach(() => {
        myComponent.find('[name="chatName"]').simulate('change', {
            target: {
                value : newTestChatName,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().name).toEqual(newTestChatName);
    })

})

describe('when push the button', () => {
    
    const mockCurrentUserId = '0'     
    const mockOnSubmit = jest.fn()

    const myComponent = shallow(<NewChatForm
        onSubmitNewChat = { mockOnSubmit } currentUserId = { mockCurrentUserId }
    />)

    beforeEach(() => {
        myComponent.find('form').simulate('submit', {
            preventDefault: () => { },
        })
    })

    it('calls the props.onSubmit', () => {
        expect(mockOnSubmit).toHaveBeenCalledTimes(1);
    })

})