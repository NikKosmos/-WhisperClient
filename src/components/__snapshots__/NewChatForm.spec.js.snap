// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewChatForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Create new chat
  </h1>
  <input
    name="chatName"
    onChange={[Function]}
    placeholder="chat name"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Create
  </button>
</form>
`;
